src/rgw/rgw_sal_motr.cc

```diff
@@ -2423,6 +2476,9 @@ int MotrMultipartUpload::delete_parts(const DoutPrefixProvider *dpp)
       std::unique_ptr<rgw::sal::Object> obj;
       obj = this->bucket->get_object(rgw_obj_key(part_obj_name));
       std::unique_ptr<rgw::sal::MotrObject> mobj(static_cast<rgw::sal::MotrObject *>(obj.release()));
+      // As this is part of multipart object, set category to MultiMeta
+      mobj->set_category(RGWObjCategory::MultiMeta);
+      mobj->meta = mmpart->meta;
```

Comment:
This code (setting meta of object) was missing earlier. Due to this, when "open_mobj()" was called, it used to read this meta from bucket index, rather than reading it from object's part index. There are no entries for object in bucket index when object is still in multipart session (i.e. Multipart Complete is not called).

Reply:
Good catch!